pkg/controllers/clusterresourceplacement/controller_test.go

```diff
@@ -2017,6 +2079,57 @@ func TestGetOrCreateClusterResourceSnapshot_failure(t *testing.T) {
 	}
 }
 
+func TestSplitSelectedResources(t *testing.T) {
+	serviceResourceContent := *serviceResourceContentForTest(t)
+	deploymentResourceContent := *deploymentResourceContentForTest(t)
+	secretResourceContent := *secretResourceContentForTest(t)
+	tests := []struct {
+		name                       string
+		selectedResourcesSizeLimit int
+		selectedResources          []fleetv1beta1.ResourceContent
+		wantSplitSelectedResources [][]fleetv1beta1.ResourceContent
+	}{
+		{
+			name:                       "empty selected resources",
+			selectedResources:          []fleetv1beta1.ResourceContent{},
+			wantSplitSelectedResources: nil,
+		},
+		{
+			name:                       "selected resources don't cross individual clusterResourceSnapshot size limit",
+			selectedResourcesSizeLimit: 1000,
+			selectedResources:          []fleetv1beta1.ResourceContent{secretResourceContent, serviceResourceContent, deploymentResourceContent},
+			wantSplitSelectedResources: [][]fleetv1beta1.ResourceContent{{secretResourceContent, serviceResourceContent, deploymentResourceContent}},
+		},
+		{
+			name:                       "selected resources cross individual clusterResourceSnapshot size limit - each resource in separate list",
+			selectedResourcesSizeLimit: 500,
+			selectedResources:          []fleetv1beta1.ResourceContent{secretResourceContent, serviceResourceContent, deploymentResourceContent},
+			wantSplitSelectedResources: [][]fleetv1beta1.ResourceContent{{secretResourceContent}, {serviceResourceContent}, {deploymentResourceContent}},
+		},
+		{
+			name:                       "selected resources cross individual clusterResourceSnapshot size limit - two resources in first list, one resource in second list",
+			selectedResourcesSizeLimit: 600,
+			selectedResources:          []fleetv1beta1.ResourceContent{secretResourceContent, serviceResourceContent, deploymentResourceContent},
+			wantSplitSelectedResources: [][]fleetv1beta1.ResourceContent{{secretResourceContent, serviceResourceContent}, {deploymentResourceContent}},
+		},
+		{
+			name:                       "selected resources cross individual clusterResourceSnapshot size limit - one resource in first list, two resources in second list",
+			selectedResourcesSizeLimit: 600,
+			selectedResources:          []fleetv1beta1.ResourceContent{serviceResourceContent, deploymentResourceContent, secretResourceContent},
+			wantSplitSelectedResources: [][]fleetv1beta1.ResourceContent{{serviceResourceContent}, {deploymentResourceContent, secretResourceContent}},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			resourceSnapshotResourceSizeLimit = tc.selectedResourcesSizeLimit
```

Comment:
we need to revert the value to the original value at the end.

```suggestion
        originalResourceSnapshotResourceSizeLimit := resourceSnapshotResourceSizeLimit
        defer func() {
           resourceSnapshotResourceSizeLimit = originalResourceSnapshotResourceSizeLimit 
        }
	for _, tc := range tests {
		t.Run(tc.name, func(t *testing.T) {
			resourceSnapshotResourceSizeLimit = tc.selectedResourcesSizeLimit
```